cloud/scope/powervs_image.go

```diff
@@ -130,8 +145,20 @@ func NewPowerVSImageScope(params PowerVSImageScopeParams) (scope *PowerVSImageSc
 		},
 		CloudInstanceID: spec.ServiceInstanceID,
 	}
-	c, err := powervs.NewService(options)
 
+	if params.IBMPowerVSCluster == nil {
+		err = errors.New("failed to generate new scope from nil IBMPowerVSCluster")
+		return
+	}
+	scope.IBMPowerVSCluster = params.IBMPowerVSCluster
+
+	for _, powervsEndpoint := range params.ServiceEndpoint {
```

Comment:
and also wondering if we can move this function to endpoint package as utility function so that can be reused everywhere?

Reply:
and also print add some debug log for the override